std/conv.d

```diff
@@ -4984,1040 +4984,8 @@ private bool isOctalLiteral(const string num)
-    __std_conv_SS ss = __std_conv_SS(1);
-
-    __std_conv_S sTest1 = ss; //this calls "SS alias this" (and not "S.this(SS)")
-    emplace(&s, ss); //"alias this" should take precedence in emplace over "opCall"
-    assert(s.i == 1);
-}
-
-//Nested classes
-@system unittest
-{
-    class A{}
-    static struct S
-    {
-        A a;
-    }
-    S s1 = void;
-    S s2 = S(new A);
-    emplace(&s1, s2);
-    assert(s1.a is s2.a);
-}
-
-//safety & nothrow & CTFE
-@system unittest
-{
-    //emplace should be safe for anything with no elaborate opassign
-    static struct S1
-    {
-        int i;
-    }
-    static struct S2
-    {
-        int i;
-        this(int j)@safe nothrow{i = j;}
-    }
-
-    int i;
-    S1 s1 = void;
-    S2 s2 = void;
-
-    auto pi = &i;
-    auto ps1 = &s1;
-    auto ps2 = &s2;
-
-    void foo() @safe nothrow
-    {
-        emplace(pi);
-        emplace(pi, 5);
-        emplace(ps1);
-        emplace(ps1, 5);
-        emplace(ps1, S1.init);
-        emplace(ps2);
-        emplace(ps2, 5);
-        emplace(ps2, S2.init);
-    }
-    foo();
-
-    T bar(T)() @property
-    {
-        T t/+ = void+/; //CTFE void illegal
-        emplace(&t, 5);
-        return t;
-    }
-    // CTFE
-    enum a = bar!int;
-    static assert(a == 5);
-    enum b = bar!S1;
-    static assert(b.i == 5);
-    enum c = bar!S2;
-    static assert(c.i == 5);
-    // runtime
-    auto aa = bar!int;
-    assert(aa == 5);
-    auto bb = bar!S1;
-    assert(bb.i == 5);
-    auto cc = bar!S2;
-    assert(cc.i == 5);
-}
-
-
-@system unittest
-{
-    struct S
-    {
-        int[2] get(){return [1, 2];}
-        alias get this;
-    }
-    struct SS
-    {
-        int[2] ii;
-    }
-    struct ISS
-    {
-        int[2] ii;
-    }
-    S s;
-    SS ss = void;
-    ISS iss = void;
-    emplace(&ss, s);
-    emplace(&iss, s);
-    assert(ss.ii == [1, 2]);
-    assert(iss.ii == [1, 2]);
-}
-
-//disable opAssign
-@system unittest
-{
-    static struct S
-    {
-        @disable void opAssign(S);
-    }
-    S s;
-    emplace(&s, S.init);
-}
-
-//opCall
-@system unittest
-{
-    int i;
-    //Without constructor
-    {
-        static struct S1
-        {
-            int i;
-            static S1 opCall(int*){assert(0);}
-        }
-        S1 s = void;
-        static assert(!__traits(compiles, emplace(&s,  1)));
-    }
-    //With constructor
-    {
-        static struct S2
-        {
-            int i = 0;
-            static S2 opCall(int*){assert(0);}
-            static S2 opCall(int){assert(0);}
-            this(int i){this.i = i;}
-        }
-        S2 s = void;
-        emplace(&s,  1);
-        assert(s.i == 1);
-    }
-    //With postblit ambiguity
-    {
-        static struct S3
-        {
-            int i = 0;
-            static S3 opCall(ref S3){assert(0);}
-        }
-        S3 s = void;
-        emplace(&s, S3.init);
-    }
-}
+// emplace() used to be here but was moved to druntime
+public import core.lifetime : emplace;
```

Comment:
Open for better suggestions...

Reply:
I think it would be best to issue a deprecation that suggests using `import core.lifetime`/`core.inernal.lifetime` instead of `std.conv` when using `emplace`/`emplaceRef`. That would also require adding a changelog entry.